File: 10. Week 10 - Interaction and Animation/exercise/python/visualization.py
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import os

def visualize():
    data_path = '../../data/linked_data.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    # TODO: Load Data
    # df = ...
    
    # Setup Figure with Subplots
    # TODO: Create subplots with 1 row and 2 columns
    # fig = make_subplots(...)
    
    # --- 1. Scatter Plot ---
    # TODO: Create a scatter plot trace
    # scatter = go.Scatter(...)
    # TODO: Add trace to figure
    # fig.add_trace(scatter, row=1, col=1)
    
    # --- 2. Bar Chart (Aggregated) ---
    # TODO: Aggregate data for bar chart
    # monthly = ...
    
    # TODO: Create a bar chart trace
    # bar = go.Bar(...)
    # TODO: Add trace to figure
    # fig.add_trace(bar, row=1, col=2)
    
    # Layout
    # TODO: Update layout with title and sizing
    # fig.update_layout(...)
    
    # Axis labels
    # TODO: Update axis labels
    # fig.update_xaxes(...)
    # fig.update_yaxes(...)
    
    # Save
    # TODO: Write to HTML
    # fig.write_html("interactive_dashboard.html")
    
    print("Interactive Dashboard saved to interactive_dashboard.html")

if __name__ == "__main__":
    visualize()

File: 10. Week 10 - Interaction and Animation/exercise/r/visualization.R
# Visualize Interactive Dashboard using ggplot2 and plotly

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
if (!require(plotly)) install.packages("plotly")
library(ggplot2)
library(readr)
library(plotly)

data_path <- "../../data/linked_data.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # 1. Create Base ggplot
  # Scatter Plot of Distance vs Pace with tooltip text
  # TODO: Create the base ggplot with a 'text' aesthetic for tooltips
  
  # p <- ggplot(df, aes(x = Distance_km, y = Pace_min_km, 
  #                     text = ...)) + 
  #   geom_point(...)
  
  
  # 2. Convert to Plotly for Interaction
  # TODO: Use ggplotly() to convert 'p' to an interactive plot
  
  # interactive_plot <- ggplotly(...)
  
  
  # 3. Save as HTML
  # if (!require(htmlwidgets)) install.packages("htmlwidgets")
  # library(htmlwidgets)
  
  # saveWidget(interactive_plot, "interactive_dashboard.html", selfcontained = TRUE)
  # print("Interactive Dashboard saved to interactive_dashboard.html")

} else {
  print(paste("Error: File not found at", data_path))
}
